### PR TITLE
timestamp logs for entry/exit for all the top-level Lifecycle package…

### PR DESCRIPTION
**Pull request by joe-kimmel-vmw**
… functions

See discussion in https://github.com/buildpacks/lifecycle/issues/1153

### CODE REVIEW

File: log/timelog_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+func (m mockLog) Error(msg string) {
+	m.incr("Error")
+}
+func (m mockLog) Errorf(fmt string, v ...interface{}) {
+	m.incr("Error")
+}
+
+func TestCache(t *testing.T) {
+	spec.Run(t, "Exporter", testCache, spec.Parallel(), spec.Report(report.Terminal{}))
```
Comment (dlion) [inline log/timelog_test.go:53]: The name of the func and the run descriptor should be changed to `TestTimelog` and `TImelog` or something around that 😄

File: log/timelog_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func testCache(t *testing.T, when spec.G, it spec.S) {
+	when("#it's like this", func() {
+		it("it's like that", func() {
```
Comment (dlion) [inline log/timelog_test.go:58]: Same thing here for the name of the function and when/it description 😄

**Comment by dlion** [comment]
Thanks Joe! I added a few comments 😄

File: analyzer.go [hunk 3]
```diff
@@ -190,6 +190,8 @@ func (f *AnalyzerFactory) setRun(analyzer *Analyzer, imageRef string) error {
 
 // Analyze fetches the layers metadata from the previous image and writes analyzed.toml.
 func (a *Analyzer) Analyze() (files.Analyzed, error) {
+	ft := log.NewFuncTimer("Analyzer", a.Logger)
```
Comment (jabrown85) [inline analyzer.go:193]: I think the 2 step approach of creating the timer and manually deferring is good to have, but maybe a convenience method would be nice for the most common approach?
```suggestion
	defer log.MeasureDuration("Analyzer", a.Logger)
```

Comment (joe-kimmel-vmw) [reply analyzer.go:193]: Thanks @jabrown85  - The `New` function is actually also writing a Start timestamp to the logs, so maybe I should call it `NewAndLog` or similar?   There is a 3-step version available if you (1) make the struct yourself, (2) call `RecordStart` and then do  ` defer RecordEnd()`

Comment (joe-kimmel-vmw) [reply analyzer.go:193]: or in other words,  I'm not sure there's a one-liner that outputs both a start and end log timestamp

Comment (jabrown85) [reply analyzer.go]: Ah, right. I think a method could be written like this

```
func MeasureDuration(msg string) {
  ft := log.NewRecordStart(msg, b.Logger)
  return ft.RecordEnd()
}

// caller
defer log.MeasureDuration("Analyzer", a.Logger)() <-- notice second function is deferred
```

Comment (joe-kimmel-vmw) [reply analyzer.go]: indeed it can! :)

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**